### PR TITLE
Shooting NPCs and more

### PR DESCRIPTION
**Pull request by metalgearsloth**
Kinda scope-creeping to cleanup stuff I wasn't entirely happy about.

Fixes https://github.com/space-wizards/space-station-14/issues/18205

- Don't make primitive tasks prototypes as it's kind of a pain to when you have to reference other areas.
- Split out melee and jukes.
- Basic Shooting / pickups / reloading.
- The umpteenth time I touch collision avoidance.
- Fix some pathfinding shit.
- Fix a lot of other random shit I found with NPCs.
- Probably going to make blackboard shit enums as the constant strings probably aren't worth it for forks (laterTM).

https://github.com/space-wizards/space-station-14/assets/31366439/f71cd4d3-aeb8-437a-86fe-a768703728fb

Known issues:
- Current target stuff is scrungly bungly but needs NPC memory.
- The scope crept a lot so I left some stuff as is.

:cl:
- add: Added xeno spitters to salvage.
- fix: Fix dragon carps not looking like their dragon.
- fix: Fix dragon carp NPCs not following the dragon and not idling on spawn.
- fix: Fix NPCs not shutting down properly upon player takeover.

### CODE REVIEW

**Comment by Just-a-Unity-Dev** [comment]
the one thing frigid needed.......

File: Resources/Prototypes/NPCs/utility_queries.yml [hunk 1]
```diff
@@ -17,6 +34,23 @@
...
 
+- type: utilityQuery
+  id: NearbyGuns
+  query:
+    - !type:ComponentQuery
+      components:
+        - type: Gun
+        - type: Item
+  considerations:
+    # TODO: Prefer highest DPC probably?
```
Comment (moonheart08) [inline Resources/Prototypes/NPCs/utility_queries.yml:45]: Should probably find some way to do multivariable optimization so it'll also consider how much ammo is available.

File: Content.Server/NPC/NPCBlackboard.cs [hunk 2]
```diff
@@ -168,7 +198,53 @@ private bool TryGetEntityDefault(string key, [NotNullWhen(true)] out object? val
...
+                {
+                    if (!hand.IsEmpty)
+                        continue;
+
+                    handos.Add(id);
+                }
+
+                value = handos;
+                return true;
+            }
```
Comment (moonheart08) [inline Content.Server/NPC/NPCBlackboard.cs:245]: this whole switch is starting to look suspiciously like copy paste, can this be deduplicated somehow?

**Review by moonheart08 — APPROVED** [review]
here's ur hom,ework

**Pull request merged**